gcloud-java-dns/src/main/java/com/google/gcloud/dns/ManagedZoneInfo.java

```diff
@@ -0,0 +1,334 @@
+     */
+    Builder creationTimeMillis(long creationTimeMillis) {
+      checkArgument(creationTimeMillis >= 0, "The timestamp cannot be negative.");
+      this.creationTimeMillis = creationTimeMillis;
+      return this;
+    }
+
+    /**
+     * Sets a mandatory DNS name of this managed zone, for instance "example.com.".
+     */
+    public Builder dnsName(String dnsName) {
+      this.dnsName = checkNotNull(dnsName);
+      return this;
+    }
+
+    /**
+     * Sets a mandatory description for this managed zone. The value is a string of at most 1024
+     * characters (this limit is posed by Google Cloud DNS; gcloud-java does not enforce the limit)
+     * which has no effect on the managed zone's function.
+     */
+    public Builder description(String description) {
+      this.description = checkNotNull(description);
+      return this;
+    }
+
+    /**
+     * Optionally specifies the NameServerSet for this managed zone. A NameServerSet is a set of DNS
+     * name servers that all host the same ManagedZones.
+     */
+    public Builder nameServerSet(String nameServerSet) {
+      // todo(mderka) add more to the doc when questions are answered by the service owner
+      this.nameServerSet = checkNotNull(nameServerSet);
+      return this;
+    }
+
+    /**
+     * Sets a list of servers that hold the information about the managed zone. This information is
+     * provided by Google Cloud DNS and is read only.
+     */
+    Builder nameServers(List<String> nameServers) {
+      this.nameServers.addAll(checkNotNull(nameServers));
+      return this;
+    }
+
+    /**
+     * Removes all the nameservers from the list.
+     */
+    Builder clearNameServers() {
+      this.nameServers.clear();
+      return this;
+    }
+
+    /**
+     * Builds the instance of ManagedZoneInfo based on the information set here.
+     */
+    public ManagedZoneInfo build() {
+      return new ManagedZoneInfo(this);
+    }
+  }
+
+  private ManagedZoneInfo(Builder builder) {
+    this.name = builder.name;
+    this.id = builder.id;
+    this.creationTimeMillis = builder.creationTimeMillis;
+    this.dnsName = builder.dnsName;
+    this.description = builder.description;
+    this.nameServerSet = builder.nameServerSet;
+    this.nameServers = ImmutableList.copyOf(builder.nameServers);
+  }
+
+  /**
+   * Returns a builder for {@code ManagedZoneInfo} with an assigned {@code name}.
+   */
+  public static Builder builder(String name) {
+    return new Builder(name);
+  }
+
+  /**
+   * Returns a builder for {@code ManagedZoneInfo} with an assigned {@code id}.
+   */
+  public static Builder builder(Long id) {
+    return new Builder(id);
+  }
+
+  /**
+   * Returns a builder for {@code ManagedZoneInfo} with an assigned {@code name} and {@code id}.
+   */
+  public static Builder builder(String name, Long id) {
+    return new Builder(name, id);
+  }
+
+  /**
+   * Returns an empty builder for {@code ManagedZoneInfo}. We use it internally in {@code toPb()}.
```

Comment:
Maybe move the comment "We use it internally in {@code toPb()}." to the constructor (and you may call it directly).